src/mongo/client/dbclient.cpp

```diff
@@ -1365,6 +1366,13 @@ void DBClientWithCommands::_findAndModify(const StringData& ns,
     commandBuilder.append("new", returnNew);
     commandBuilder.append("upsert", upsert);
 
+    if (getMaxWireVersion() >= 4) {
```

Comment:
I feel like ignoring a user requested writeConcern is dangerous. I think we should take the suggestion from @amidvidy